### ISSUE TITLE
Tests of example code in docs

### ISSUE THREAD

**Issue by AdamIsrael** [juju-solutions/charms.reactive#115]
It's a daunting task to go through docs and test every block of example code to see if it still applies to the current release. As the library evolves, the examples become stale and may break. I'd like to see if we can find a way to include testing these examples as part of our review process.

It's not a unique problem. There may be a way to include code in the doc that lives elsewhere in the project, like unit tests, that are run in CI/CD and updates to them are automatically reflected in the docs.

tl;dr: It's be cool to test example code as part of a CI/CD pipeline so a committer knows if the patch they just landed breaks what's provided by the docs.
